### PR TITLE
Gpio example

### PR DESCRIPTION
**Pull request by NOP0**
FWIW we're able to read a GPIO on a Raspberry PI 2B in real life. Writing is not done/tested yet, I have to get some more hardware. This is a branch of #29 , I'm not sure if you prefer to merge that first?

### CODE REVIEW

**Comment by Michael-F-Bryan** [comment]
> This is a branch of #29, I'm not sure if you prefer to merge that first?

I just did a rebase and merge of #29 so this branch should be able to make progress. Do you think it's worth testing on real hardware before merging? Or could we hit merge without it?

**Comment by NOP0** [comment]
>Do you think it's worth testing on real hardware before merging? Or could we hit merge without it?

Started working on gpio output, but just realized that outputs are not implemented in ```ProcessImage``` Ooops :laughing: 

I'll need to implement that first.

**Comment by NOP0** [comment]
@Michael-F-Bryan  Ok, did some more on this...The gpio-example now logs TRUE∕FALSE when I put voltage on the input pin, and I can measure a toggling voltage on the output pin. Do you have access to a Raspberry PI, should we double-check this?

File: core/src/process_image.rs [hunk 1]
```diff
@@ -84,29 +92,71 @@ impl ProcessImage {
...
-                AccessType::Word => self.write_word(
-                    device.1.byte,
-                    devices.read::<u16>(device.0).unwrap(),
-                ),
-                AccessType::DoubleWord => self.write_double_word(
-                    device.1.byte,
-                    devices.read::<u32>(device.0).unwrap(),
-                ),
-            }
+        match self.direction {
```
Comment (NOP0) [inline core/src/process_image.rs]: Is this approach ok?

**Comment by NOP0** [comment]
I've  read a bit on how the embedded guys do this stuff; https://rust-embedded.github.io/book/static-guarantees/design-contracts.html 

Their approach might be better, thinking of redoing this like in the link. Thoughts?

I have looked a bit at the embedded-hal traits also, and for gpio I believe we can be "conformant" without any hassle. I'm on my phone, so no code to show for though.

**Comment by NOP0** [comment]
@Michael-F-Bryan 
Rewrote process image to use proper types instead of "DoubleWord" and so on. 

What do you think of c0f1bdc ? Is it ok to match on adress, or should this be more generic, so that any type which implements ```PiAccess``` could update the process image? I'm not sure how I would do that though :smile:

**Pull request merged**